diagnostic.md

```diff
@@ -3,36 +3,42 @@
 What are three advantages of using Ember?
 
 ```sh
-# your answer here
+1. Unlike some front-end javascript frameworks, URL routing is central to Ember.  URL routes are used to transition state, fetch data, and in some cases, replicate browser history.  Because of this, ember applications can be said to more fully embrace the web.
+2. Ember is a mature framework with a lot of tooling supporting it.  The Ember CLI, for example, allows developers to quickly scaffold new applications and add new features to existing applictions.
+3. A developer who understands ember can more easily begin working on an existing application because of the frameworks adherence to convention.
 ```
 
 What is the templating system used in Ember? How it is related to
 handlebars?
 
-```sh
-# your answer here
+```md
+Ember templates are simply handlebars files.  The only difference between normal handlebars and ember handlebars (and I believe this is not exposed to framework users in any meaningful way) is that ember's implementation allows  for access to Ember Object properties.
 ```
 
 Name at least two layers of an Ember application (1.0 or 2.0):
 
 ```sh
-# your answer here
+Routing Layer
+Model Layer
 ```
 
 How does a component differ from an Ember template?
 
 ```sh
-# your answer here
+Components in ember are an implmentation of the web component standard.  They render to custom-named HTML elements in the DOM.  Conceptually components differ from templates in that components should be stand-alone, self contained units not tied to a given route or model.
```

Comment:
Although Ember is on-track to have custom-named DOM elements as soon as they're available, it can't do that yet. Components contain re-usable functionality tied to a template.
.5/1